auto_gptq/modeling/_base.py

```diff
@@ -865,6 +929,7 @@ def from_quantized(
         if use_safetensors:
             extensions.append(".safetensors")
         else:
+            logger.warning("Deprecation: Loading of non-safetensor weights will be removed in the future. Please convert your weights to safetensor format. ")
```

Comment:
Why would it be?

Reply:
Reason for the change is that I don't believe there is a non-safetensor future based on this: 

https://github.com/huggingface/transformers/pull/29846/files
and comment/reason behind the PR here: https://github.com/huggingface/transformers/issues/29500#issuecomment-1983289037

> This is to continue pushing safetensors forward as an alternative to raw .bin checkpoints as we see more bad actors with pure .bin checkpoints. Thanks again for your report!

HF is taking very aggressive steps to speed up safetensor adoption due to threat-actors using .bin files. I am not privy to hf security stats but this opened my eyeball as forcing safetensor usage in the long run.

Reply:
Let's discuss this in an other PR/issue, so as to keep this one light if possible.

Although safetensors is more widely adopted (and the default in HF libs), for now most (all?) libraries still support .bin recommended by PyTorch, so let's support as well (although not really tested in AutoGPTQ tests).

AFAIK there is no plan to support safetensors in PyTorch natively